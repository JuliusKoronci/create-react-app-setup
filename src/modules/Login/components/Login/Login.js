/* eslint-disable no-underscore-dangle */
// @flow
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Login extends PureComponent {
  static defaultProps: Object;
  static propTypes: Object;

  render() {
    const { formValues, handleChange, handleSubmit } = this.props;
    return (
      <form>
        <input
          type="text"
          value={formValues._username}
          placeholder="Username"
          onChange={e => handleChange('_username', e.target.value)}
        />
        <input
          type="text"
          value={formValues._password}
          placeholder="Password"
          onChange={e => handleChange('_password', e.target.value)}
        />
        <button type="submit" onClick={handleSubmit}>Submit</button>
      </form>
    );
  }
}

Login.defaultProps = {
  formValues: {
    _username: '',
    _password: '',
  },
};
Login.propTypes = {
  formValues: PropTypes.shape({
    _username: PropTypes.string,
    _password: PropTypes.string,
  }).isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Login;

