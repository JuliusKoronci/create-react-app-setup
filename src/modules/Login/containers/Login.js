// @flow
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import LoginView from '../components/Login';
import {} from '../actions/LoginActions';

export class Login extends Component {

	render() {
		return (
			<LoginView />
		);
	}
}

Login.propTypes = {
	test: PropTypes.string,
};

function _mapStoreToProps(state, ownProps) {
	return {};
}
function _mapDispatchToProps(dispatch) {
	return bindActionCreators({}, dispatch);
}

export default connect(_mapStoreToProps, _mapDispatchToProps)(Login);

