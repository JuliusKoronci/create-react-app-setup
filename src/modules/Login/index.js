import loginReducer from './reducers/LoginReducer';
export { loginReducer };
