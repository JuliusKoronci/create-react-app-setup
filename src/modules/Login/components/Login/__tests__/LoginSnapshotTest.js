import React from 'react';
import Login from '../Login';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const wrapper = shallow(<Login />);
describe('(Component) Login', () => {
	it('Should match snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});

