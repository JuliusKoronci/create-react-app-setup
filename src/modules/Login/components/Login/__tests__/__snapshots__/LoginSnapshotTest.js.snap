// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Login Should match snapshot 1`] = `
<div>
  Login
</div>
`;
